README.md

```diff
@@ -218,6 +221,12 @@ If the SASL_OAUTHBEARER listener is also used for interbroker communication, the
 
     listener.name.client.oauthbearer.sasl.login.callback.handler.class=io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler 
 
+#### Enabling the custom principal builder
+
+OAuth authentication also requires a custom principal builder to be installed on the broker:
+  
+    principal.builder.class=io.strimzi.kafka.oauth.server.OAuthKafkaPrincipalBuilder
```

Comment:
How does this work for other listener using different authentication methods then OAuth?

Reply:
That's a good question. Broker to broker via mTLS still works. SASL_PLAIN works under the assumption that it's OAuth over PLAIN, I haven't tried with SASL_SCRAM but it's possible that it wouldn't work.
But the idea is to have a single store of users, isn't it? Do we really want to support one set of users on one listener, and a completely different unrelated set of users on another listener, and they both are then subject to a single global authorizer?

Reply:
> Do we really want to support one set of users on one listener, and a completely different unrelated set of users on another listener, and they both are then subject to a single global authorizer?

I think we do want to support it. there might be different groups of users - for example cluster operators with one auth mechanism on one interface, internal applications external applications etc. So there are several concerns:
* The authentication used internally for replication (today mTLS, but might be SASL in the future since there are many request to be able to avoid using TLS etc.)
* The authentication on different listeners.

The first one is a must have - that has to work. The other one is IMHO very important. But if it doesn't work we might be able to live with it but needs to be properly documented. There is also backwards compatibility to keep in mind. It is not clear to me if this is needed only for OAuth over PLAIN which is a new feature and users do not have to enable it. Or whether this affects also the OAUTHBEARER users who already might be using it in combination with other authentication methods. You could possibly also investigate if this can be configured on per-listener basis or not => as that might be a solution of course since it would not impact other listeners.

Reply:
The way OAuthKafkaPrincipalBuilder class is written it delegates to the DefaultKafkaPrincipalBuilder class if none of the OAuth based credentials are found. In principle it should work for any different co-existing SASL mechanisms.
It's just that there is no test for it, so it might as well not work.

Reply:
I think you should test it then. Because we need to know how it does or does not work.

Reply:
I added the test for this to `refactored-testsuite` branch (#85). Multiple SASL mechanisms on multiple listeners appear to work without any issues.

Reply:
The tests are passing for #85.